components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/WorkflowsApiServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+        String username = RestApiUtil.getLoggedInUsername();
+        try {
+
+            APIStore apiStore = RestApiUtil.getConsumer(username);
+            Workflow workflow = apiStore.retrieveWorkflow(workflowReferenceId);
+            if (workflow == null) {
+                String errorMessage = "Workflow entry not found for: " + workflowReferenceId;
+                APIMgtResourceNotFoundException e = new APIMgtResourceNotFoundException(errorMessage,
+                        ExceptionCodes.WORKFLOW_NOT_FOUND);
+                HashMap<String, String> paramList = new HashMap<String, String>();
+                paramList.put(APIMgtConstants.ExceptionsConstants.WORKFLOW_REF_ID, workflowReferenceId);
+                ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getErrorHandler(), paramList);
+                log.error(errorMessage, e);
+                return Response.status(e.getErrorHandler().getHttpStatusCode()).entity(errorDTO).build();
+            } else if (WorkflowStatus.APPROVED == workflow.getStatus()) {
+                String errorMessage = "Workflow is already in complete state";
+                APIMgtResourceNotFoundException e = new APIMgtResourceNotFoundException(errorMessage,
+                        ExceptionCodes.WORKFLOW_COMPLETED);
+                HashMap<String, String> paramList = new HashMap<String, String>();
+                paramList.put(APIMgtConstants.ExceptionsConstants.WORKFLOW_REF_ID, workflowReferenceId);
+                ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getErrorHandler(), paramList);
+                log.error(errorMessage, e);
+                return Response.status(e.getErrorHandler().getHttpStatusCode()).entity(errorDTO).build();
+
+            } else {
+                WorkflowExecutor workflowExecutor = WorkflowExecutorFactory.getInstance()
+                        .getWorkflowExecutor(workflow.getWorkflowType());
+
+                if (body == null) {
+                    RestApiUtil.handleBadRequest("Request payload is missing", log);
+                }
+
+                if (body.getDescription() != null) {
+                    workflow.setWorkflowDescription(body.getDescription());
+                }
+
+                if (body.getStatus() == null) {
+                    RestApiUtil.handleBadRequest("Workflow status is not defined", log);
+                } else {
+                    workflow.setStatus(WorkflowStatus.valueOf(body.getStatus().toString()));
+                }
+
+                if (body.getAttributes() != null) {
+                    workflow.setAttributes(body.getAttributes());
+                }
+
+                apiStore.completeWorkflow(workflowExecutor, workflow);
+                return Response.ok().entity(body).build();
+            }
+
+        } catch (APIManagementException e) {
+            String errorMessage = "Error while completing workflow for reference : " + workflowReferenceId;
+            HashMap<String, String> paramList = new HashMap<>();
```

Comment:
RestApiUtil.getErrorDTO() method uses hashmap instead of map as a param. So used a hashmap